src/components/Signup.js

```diff
@@ -0,0 +1,216 @@
+                          <ul className="form-instrcutions">
+                                <li><p>Make sure no fields are empty when user submits the form. </p></li>
+                                <li><p>Make sure email address entered is always valid.</p></li>
+                                <li><p>Password should follow following criteria:<br />
+                                    a. Minimum length 6 characters. <br />
+                                    b. Must have a number and alphabet.<br />
+                                    c. Cannot have spaces.<br /></p>
+                                </li>
+                                <li><p>Confirm password needs to be same as password value.</p></li>
+                          </ul>
+                      </div>) }
+                  </div>
+                </div>
+                <div> </div>
+              </div>
+            </div>
+            <div className="col span-1-of-2">
+                  <form method="" action="" onSubmit={this.onSubmitHandler}>  
+                  <div className="floating-input-area">
+                    <input type="text" aria-label={Constants.TEXTS.FORM_INPUTS.fname.label} className="inputText" onChange={(event)=>this.inputHandler(event,Constants.TEXTS.SIGNUP.form_values.fname)} placeholder=" " required/>
+                    <span className="floating-label">{Constants.TEXTS.FORM_INPUTS.fname.value}</span>
+                  </div>
+                  <div className="floating-input-area">
+                    <input type="text" aria-label={Constants.TEXTS.FORM_INPUTS.lname.label} className="inputText" onChange={(event)=>this.inputHandler(event,Constants.TEXTS.SIGNUP.form_values.lname)} placeholder=" " required/>
+                    <span className="floating-label">{Constants.TEXTS.FORM_INPUTS.lname.value}</span>
+                  </div>             
+                   <div className="floating-input-area">
+                    <input type="email" aria-label={Constants.TEXTS.FORM_INPUTS.email.label} className="inputText" onChange={(event)=>this.inputHandler(event,Constants.TEXTS.SIGNUP.form_values.email)} placeholder=" " required/>
+                    <span className="floating-label">{Constants.TEXTS.FORM_INPUTS.email.value}</span>
+                  </div>
+                  <div className="floating-input-area">
+                    <input type="password" aria-label={Constants.TEXTS.FORM_INPUTS.pwd.label} className="inputText" onChange={(event)=>this.inputHandler(event,Constants.TEXTS.SIGNUP.form_values.pwd)} placeholder=" " required/>
+                    <span className="floating-label">{Constants.TEXTS.FORM_INPUTS.pwd.value}</span>
+                  </div>
+                  <div className="floating-input-area">
+                    <input type="password" aria-label={Constants.TEXTS.FORM_INPUTS.cpwd.label} className="inputText" onChange={(event)=>this.inputHandler(event,Constants.TEXTS.SIGNUP.form_values.cpwd)} placeholder=" " required/>
```

Comment:
not getting the idea of passing a constant as a second argument to onChange handler. If you want to identify the element that triggered the event, use "name" instead which is already available on the event object

Reply:
I created form values for each input at signup in constants and in the component there is a function that does action based on the constant value of particular input. for all inputs, i want to use only one function by passing a particular input name.